sklearn/tree/_criterion.pyx

```diff
@@ -84,8 +104,23 @@ cdef class Criterion:
 
         pass
 
+    cdef void reset_without_missing(self) nogil:
+        """Reset the criterion at pos=start. Leave the missing samples unaffected.
+
+        This method must be implemented by the subclass.
+        """
+
+        pass
+
     cdef void reverse_reset(self) nogil:
-        """Reset the criterion at pos=end.
+        """Reset the criterion at pos=end(-n_missing).
+
+        This method must be implemented by the subclass.
+        """
+        pass
+
+    cdef void reverse_reset_without_missing(self) nogil:
```

Comment:
Why do we need a new method for this? Couldn't you just pass an argument to reverse_reset?